From 90517a56571a1ee08bd722d3e446adba68dd9a16 Mon Sep 17 00:00:00 2001
Message-ID: <90517a56571a1ee08bd722d3e446adba68dd9a16.1736059919.git.divya@subvertising.org>
In-Reply-To: <cover.1736059918.git.divya@subvertising.org>
References: <cover.1736059918.git.divya@subvertising.org>
From: Divya Ranjan <divya@subvertising.org>
Date: Sat, 4 Jan 2025 20:58:05 +0000
Subject: [PATCH 13/44] gnu: Add rust-nvml-wrapper-0.10.

* gnu/packages/crates-io.scm (rust-nvml-wrapper-0.10): New variable.
---
 gnu/packages/crates-io.scm | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/gnu/packages/crates-io.scm b/gnu/packages/crates-io.scm
index a9c197c898..7706fd6c67 100644
--- a/gnu/packages/crates-io.scm
+++ b/gnu/packages/crates-io.scm
@@ -48049,6 +48049,34 @@ (define-public rust-numtoa-0.1
           "1vs9rhggqbql1p26x8nkha1j06wawwgb2jp5fs88b5gi7prvvy5q"))))
     (arguments '(#:tests? #f))))
 
+(define-public rust-nvml-wrapper-0.10
+  (package
+    (name "rust-nvml-wrapper")
+    (version "0.10.0")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (crate-uri "nvml-wrapper" version))
+       (file-name (string-append name "-" version ".tar.gz"))
+       (sha256
+        (base32 "1i9q80f8x5zy646wvz1wkaydryvnfnwsi8jy72hh92fll45gz6qc"))))
+    (build-system cargo-build-system)
+    (arguments
+     `(#:cargo-inputs (("rust-bitflags" ,rust-bitflags-2)
+                       ("rust-libloading" ,rust-libloading-0.8)
+                       ("rust-nvml-wrapper-sys" ,rust-nvml-wrapper-sys-0.8)
+                       ("rust-serde" ,rust-serde-1)
+                       ("rust-serde-derive" ,rust-serde-derive-1)
+                       ("rust-static-assertions" ,rust-static-assertions-1)
+                       ("rust-thiserror" ,rust-thiserror-1)
+                       ("rust-wrapcenum-derive" ,rust-wrapcenum-derive-0.4))))
+    (home-page "https://github.com/Cldfire/nvml-wrapper")
+    (synopsis
+     "safe and ergonomic Rust wrapper for the NVIDIA Management Library")
+    (description
+     "This package provides a safe and ergonomic Rust wrapper for the NVIDIA
+Management Library.")
+    (license (list license:expat license:asl2.0))))
 (define-public rust-oauth2-types-0.7
   (package
     (name "rust-oauth2-types")
-- 
2.47.1

